tools/test_utils/fullservice.py

```diff
@@ -0,0 +1,215 @@
+        print('starting full service')
+        print(cmd)
+        self.full_service_process = subprocess.Popen(cmd, shell=True)
+
+    def stop(self):
+        try:
+            subprocess.check_output("pkill full-service", shell=True)
+        except subprocess.CalledProcessError as exc:
+            if exc.returncode != 1:
+                raise
+
+    # return the result field of the request
+    def _request(self, request_data):
+        self.request_count += 1
+        print('sending request to full service')
+        url = 'http://127.0.0.1:9090/wallet'
+        default_params = {
+            "jsonrpc": "2.0",
+            "api_version": "2",
+            "id": self.request_count,
+        }
+        request_data = {**request_data, **default_params}
+
+        print(f'request data: {request_data}')
+
+        try:
+            parsed_url = urlparse(url)
+            connection = http.client.HTTPConnection(parsed_url.netloc)
+            connection.request('POST', parsed_url.path, json.dumps(request_data), {'Content-Type': 'application/json'})
+            r = connection.getresponse()
+        except ConnectionError:
+            raise ConnectionError
+
+        raw_response = None
+        try:
+            raw_response = r.read()
+            response_data = json.loads(raw_response)
+        except ValueError:
+            raise ValueError('API returned invalid JSON:', raw_response)
+        print(f'request returned {response_data}')
+
+        # TODO requests might be flakey due to timing issues... waiting 2 seconds to bypass most of these issues
+        time.sleep(2)
+        if 'error' in response_data:
+            return response_data['error']
+        else:
+            return response_data['result']
+
+    def import_account(self, mnemonic) -> bool:
+        print(f'importing full service account {mnemonic}')
+        params = {
+            'mnemonic': mnemonic,
+            'key_derivation_version': '2',
+        }
+        r = self._request({
+            "method": "import_account",
+            "params": params
+        })
+
+        if 'error' in r:
+            if 'Diesel Error: UNIQUE constraint failed' in r['error']['data']['details']:
+                return True
+            else:
+                return False
+        return True
+
+    # retrieve accounts from mobilecoin/target/sample_data/keys
+    def get_test_accounts(self) -> Tuple[str, str]:
+        print(f'retrieving accounts for account_keys_0 and account_keys_1')
+        keyfile_0 = open(os.path.join(constants.KEY_DIR, 'account_keys_0.json'))
+        keyfile_1 = open(os.path.join(constants.KEY_DIR, 'account_keys_1.json'))
+        keydata_0 = json.load(keyfile_0)
+        keydata_1 = json.load(keyfile_1)
+
+        keyfile_0.close()
+        keyfile_1.close()
+
+        return (keydata_0['mnemonic'], keydata_1['mnemonic'])
+
+    # check if full service is synced within margin
+    def sync_status(self, eps=5) -> bool:
```

Comment:
what is eps?

Reply:
How big is the ledger in these tests? Any reason to not wait for it to sync completely?

Reply:
This question is still relevant, not sure why this was marked as resolved.